### PR TITLE
Allow Jaeger remote sampler to be closed.

### PR DESCRIPTION
**Pull request by anuraaga**
When debugging logs of CI, I noticed 90% of the log is spam from leaked jaeger pollers.

### CODE REVIEW

**Review by jack-berg — APPROVED** [review]
I've noticed this too. Nice change.

File: sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/JaegerRemoteSamplerBuilder.java [hunk 1]
```diff
@@ -53,6 +54,7 @@ public JaegerRemoteSamplerBuilder setEndpoint(String endpoint) {
   public JaegerRemoteSamplerBuilder setChannel(ManagedChannel channel) {
     requireNonNull(channel, "channel");
     this.channel = channel;
+    closeChannel = false;
```
Comment (jkwatson) [inline sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/JaegerRemoteSamplerBuilder.java:57]: I think this worth some javadoc explanation.

**Review by jkwatson — APPROVED** [review]
Just needs a bit of javadoc about the non-closing of custom channels, and looks good to me.

File: sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/JaegerRemoteSamplerBuilder.java (hunk 1 above)
Comment (jkwatson) [reply sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/JaegerRemoteSamplerBuilder.java:57]: let's add javadoc as a follow-up, so we can get rid of the build spam.

**Pull request merged**